src/responses/Albany_SolutionTwoNormResponseFunction.cpp

```diff
@@ -146,3 +146,77 @@ evaluateDistParamDeriv(
     dg_dp->assign(0.0);
   }
 }
+
+void
+Albany::SolutionTwoNormResponseFunction::
+evaluateDistParamHessVecProd_xx(
+    const double current_time,
+    const Teuchos::RCP<const Thyra_MultiVector>& v,
+    const Teuchos::RCP<const Thyra_Vector>& x,
+    const Teuchos::RCP<const Thyra_Vector>& xdot,
+    const Teuchos::RCP<const Thyra_Vector>& xdotdot,
+    const Teuchos::Array<ParamVec>& param_array,
+    const Teuchos::RCP<Thyra_MultiVector>& Hv_dxdx)
+{
+  Teuchos::ScalarTraits<ST>::magnitudeType nrm = x->norm_2();
```

Comment:
You might move these inside the if statement.

Reply:
Yes that would be better! I should even include a second test which checks if the norm is not zero before using the formula.